Added extracting plural forms using standard "n"-prefixed gettext functions

vrouet opened the pull request:
This adds basic plural forms extraction, it suits my need well, it'd be great to have merged.
I'm sure there are stuff I didn't think about and that would need fixing before, so please comment on this.

vrouet commented:
I've noticed that one test if failing, but I checked on the master version and it does fail as well, so it's not something I broke ;)

BYK commented on lib/jsxgettext.js:
```diff
@@ -53,6 +53,9 @@ function getTranslatable(node, options) {
   if (options.keyword.indexOf(funcName) === -1)
     return false;
 
+  if (arg && funcName.substr(0, 1) === "n" && (isStrConcatExpr(arg) || isStringLiteral(arg)) && node.arguments[1] && (isStrConcatExpr(node.arguments[1]) || isStringLiteral(node.arguments[1])))
```
Heh, checking the first letter of the function name is a good start but may not be very good in the future. Anyways, let's add a comment above stating what this does for potential newcomers ;)

vrouet replied:
I guess here we have 2 options:
- leave it this way and explain that any gettext alias starting with "n" will be treated as a plural function
- use a strict comparison so that only "ngettext" will work. In this case there would be no way of using custom plural function until someone implements some kind of way to specify that in the calls
I'm comfortable with both these solutions, so it's your call

BYK commented on lib/jsxgettext.js:
```diff
@@ -116,7 +119,12 @@ function parse(sources, options) {
   }
 
-  options.keyword = options.keyword || ['gettext'];
+  if( options.keyword ) {
```
Looks like a combined if block is the better way to go here instead of an if block + `options.keyword || ` approach

BYK commented:
First of all this is great!  Thank you since we have been wanting this for quite some time now. I have a few requests though:

1. Please add some tests
2. Squash your commits
3. May be put an example usage/expected format somewhere since even I am not very familiar with `ngettext` usage. Adding tests may cover this.

BYK commented:
Fixes #51

vrouet commented:
Thanks for your reviews, I will definitely add some tests and doc for that, I made the pull request first basically to see if my approach was fitting your standards, and did rthings quickly because I needed to tel my dev team what to use for i18n on our project.
I will update soon

vrouet commented:
Ok, I have taken your comments into account, added some notes, integrated the plural functions to all of the relevant test cases, and squashed my commits

BYK commented:
And rebase to get tests fixed :)

vrouet commented:
Alright, rebase is done. Nice to see the whole test suite passing ;)

BYK commented on README.md:
```diff
@@ -4,9 +4,13 @@ A node module with a CLI that extracts gettext strings from JavaScript, EJS, Jad
+ngettext("Here's an apple for you", "Here are %s apples for you", 3);
+ngettext("Here's an apple" + ' for you', "Here are %s apples" + ' for you', 3);
+myModule.ngettext("Here's an apple" + ' for you', "Here are % apples" + ' for you', 3);
+ngettext.call(myObj, "Here's an apple" + ' for you', "Here are % apples" + ' for you', 3);
```
Did you mean `%s apples`? :)

BYK commented:
>Alright, rebase is done. Nice to see the whole test suite passing ;)

Definitely. You triggered me to do it :)

Final thing before merge: fix those `%s` lines in the README please?

vrouet commented:
OK now

BYK commented:
![aw yiss](http://www.awyisser.com/assets/images/thumbnail.png)
